backend/src/main/java/com/rota/api/dto/EngagementDto.java

```diff
@@ -0,0 +1,38 @@
+package com.rota.api.dto;
+
+import com.rota.database.orm.engagement.Engagement;
+import com.rota.database.orm.engagement.EngagementType;
+import io.swagger.annotations.ApiModelProperty;
+import java.time.Instant;
+import lombok.AllArgsConstructor;
+import lombok.Value;
+
+@Value
+@AllArgsConstructor
+public class EngagementDto {
+  @ApiModelProperty(value = "Staff ID")
+  int staffId;
+
+  @ApiModelProperty(value = "Starting time of an engagement")
+  Instant start;
+
+  @ApiModelProperty(value = "End time of an engagement")
+  Instant end;
+
+  @ApiModelProperty(value = "Type of engagement, e.g. SHIFT, HOLIDAY etc.")
+  EngagementType type;
```

Comment:
These model annotations are good! They are probably the ones we want. But there isn't too much point in using them for swagger unless we use them to add all the info they allow! But that can wait till another pull request when we have a firmer understanding of what we want exactly from each endpoint. Between `@ApiParam` and `@ApiModelProperty` https://www.baeldung.com/swagger-apiparam-vs-apimodelproperty